### ISSUE TITLE
[Snackbar] Provide a way to control padding

### ISSUE THREAD

**Issue by eugrro** [flutter/flutter#86627]
I was trying to figure out why there was weird padding going on shown in screenshot below.
Goal: Gain access to the entire snackbar. Currently i'm coloring it all red to show that I do not have access. I know that the background color property exists, im not trying to color it im just showing for demonstration purposes. 
```
void showError(BuildContext context, String text) {
  ScaffoldMessenger.of(context).showSnackBar(SnackBar(
    margin: EdgeInsets.only(
        bottom: Constants.bottomNavBarHeight + 10, left: MediaQuery.of(context).size.width / 6, right: MediaQuery.of(context).size.width / 6),
    behavior: SnackBarBehavior.floating,
    padding: EdgeInsets.zero,
    content: Container(
      height: 50,
      color: Colors.red,
    ),
    duration: const Duration(milliseconds: 1500),
  ));
}
```
<img src="https://user-images.githubusercontent.com/26045745/126080402-55723765-868a-48a8-8d14-f0fc6a044695.jpg"  width="300" />	

As you can see my padding is zero, so there should be no white only red.
In snack_bar.dart, line 18
```
const double _singleLineVerticalPadding = 14.0;

// TODO(ianh): We should check if the given text and actions are going to fit on
// one line or not, and if they are, use the single-line layout, and if not, use
// the multiline layout, https://github.com/flutter/flutter/issues/32782
// See https://material.io/components/snackbars#specs, 'Longer Action Text' does
// not match spec.
```
There is just a constant value here. This should be changed.
If I set _singleLineVerticalPadding = 0

<img src="https://user-images.githubusercontent.com/26045745/126080577-c11fd653-25b0-4aae-ae31-0e18b3d8dca8.jpg"  width="300" />

**Comment by eugrro** [comment]
Any updates to this? Not sure what the next steps are but this would be a useful update for me

**Comment by Piinks** [comment]
I think this was resolved by #78133
